TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,246 @@
+            this.robot.motorTwo.setPower(wheelTwoPower);
+            this.robot.motorThree.setPower(wheelThreePower); // May need to go other direction
+            this.robot.motorFour.setPower(-wheelFourPower); // May need to go other direction
+        }
+        this.telemetry.update();
+        this.stop();
+    }
+
+    public void goRight(double inches) {
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "Robot is moving right");
+        startPosition = this.robot.motorOne.getCurrentPosition();
+        endPosition = startPosition + (inches * sideInchToTick); // How far you need to travel
+        while (this.robot.motorOne.getCurrentPosition() < endPosition && this.opMode.opModeIsActive()) {
+            Log.d("goRight", "start position is " + startPosition);
+            Log.d("goRight", "end position is " + endPosition);
+            Log.d("goRight", "current position is " + this.robot.motorFour.getCurrentPosition());
+            this.robot.motorOne.setPower(wheelOnePower); // May need to turn other direct
+            this.robot.motorTwo.setPower(-wheelTwoPower); // May need to turn other direction
+            this.robot.motorThree.setPower(-wheelThreePower);
+            this.robot.motorFour.setPower(wheelFourPower);
+        }
+        this.stop();
+        this.telemetry.update();
+    }
+
+    public void stop() {
+        Log.d(DRIVE_TRAIN_CAPTION, "Stopping the drive train");
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "Stopping the drive train");
+        this.telemetry.addData("wheel power", this.robot.motorOne.getPower());
+        this.telemetry.update();
+        this.robot.motorOne.setPower(0);
+        this.robot.motorTwo.setPower(0);
+        this.robot.motorThree.setPower(0);
+        this.robot.motorFour.setPower(0);
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "Drive train is stopped");
+        this.telemetry.addData("wheel power", this.robot.motorOne.getPower());
+        this.telemetry.update();
+    }
+
+    /*public void setPower(double powerLevel){
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION,"Drive train power set");
+        wheelPower = powerLevel;
+        this.telemetry.update();
+    }*/
+    public void teleopUpdate(Gamepad gamepad1, Gamepad gamepad2) {
+        //angles = this.robot.imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES);
+        Log.d(DRIVE_TRAIN_CAPTION, "turning teleop updated");
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "gamepad updated");
+        telemetry.addData("1_left_stick_x", gamepad1.left_stick_x);
+        telemetry.addData("1_left_stick_y", gamepad1.left_stick_y);
+        telemetry.addData("1_right_stick_x", gamepad1.right_stick_x);
+        telemetry.addData("1_right_stick_y", gamepad1.right_stick_y);
+
+        double y = -gamepad1.left_stick_y; // Remember, this is reversed!
+        double x = gamepad1.left_stick_x * 1.5; // 1.5 is to counteract imperfect strafing
```

Comment:
is this part of the code that you have written this season?

Reply:
This is part of the code that I basically copy-pasted from game manual 0 earlier this season, we haven't tested it yet